src/sql/workbench/contrib/notebook/test/browser/notebookContexts.test.ts

```diff
@@ -40,90 +38,28 @@ suite('Notebook Contexts', function (): void {
 			= TypeMoq.Mock.ofType<IConnectionManagementService>(TestConnectionManagementService, TypeMoq.MockBehavior.Strict);
 
-		// No kernel or profile info provided
-		let conns = NotebookContexts.getContextsForKernel(connService.object, [mssqlProviderName]);
+		// No profile info provided
+		let conns = NotebookContexts.getContextForKernel(undefined, [mssqlProviderName]);
 		assert.deepStrictEqual(conns, defaultContext);
 
-		// No Profile, Kernels are the same
-		let kernelChangeArgs = <nb.IKernelChangedArgs>{
-			oldValue: <nb.IKernel>{
-				id: '1',
-				name: 'TestKernel'
-			},
-			newValue: <nb.IKernel>{
-				id: '1',
-				name: 'TestKernel'
-			}
-		};
-		conns = NotebookContexts.getContextsForKernel(connService.object, [mssqlProviderName], kernelChangeArgs);
-		assert.deepStrictEqual(conns, defaultContext);
-
-		// Kernel Info and Profile, but no provider IDs
+		// Profile, but no provider IDs
 		let testConn = createTestConnProfile();
-		conns = NotebookContexts.getContextsForKernel(connService.object, [], kernelChangeArgs, testConn);
-		assert.deepStrictEqual(conns, defaultContext);
+		conns = NotebookContexts.getContextForKernel(testConn, []);
+		assert.deepStrictEqual(conns, localContext);
 
 		// Normal use case
 		connService.setup(c => c.getActiveConnections()).returns(() => [testConn]);
-		conns = NotebookContexts.getContextsForKernel(connService.object, [mssqlProviderName], kernelChangeArgs, testConn);
-		assert.deepStrictEqual(conns, <IDefaultConnection>{
-			otherConnections: [testConn],
-			defaultConnection: testConn
-		});
-	});
-
-	test('Get Active Contexts', async function (): Promise<void> {
-		const connService: TypeMoq.Mock<IConnectionManagementService>
-			= TypeMoq.Mock.ofType<IConnectionManagementService>(TestConnectionManagementService, TypeMoq.MockBehavior.Strict);
-
-		let testConn = createTestConnProfile();
-
-		// No provider IDs
-		connService.setup(c => c.getActiveConnections()).returns(() => [testConn]);
-		let conns = NotebookContexts.getActiveContexts(connService.object, [], testConn);
-		assert.deepStrictEqual(conns, localContext);
-
-		// No connections
-		connService.setup(c => c.getActiveConnections()).returns(() => []);
-		conns = NotebookContexts.getActiveContexts(connService.object, [mssqlProviderName], testConn);
-		assert.deepStrictEqual(conns, defaultContext);
+		conns = NotebookContexts.getContextForKernel(testConn, [mssqlProviderName]);
+		assert.deepStrictEqual(conns, testConn);
 
-		// No valid connection IDs
-		testConn.id = '-1';
+		// Multiple provider IDs including mssql
 		connService.setup(c => c.getActiveConnections()).returns(() => [testConn]);
-		conns = NotebookContexts.getActiveContexts(connService.object, [mssqlProviderName], testConn);
-		assert.deepStrictEqual(conns, defaultContext);
+		conns = NotebookContexts.getContextForKernel(testConn, [mssqlProviderName, 'fakeProvider']);
+		assert.deepStrictEqual(conns, testConn);
```

Comment:
Would suggest adding messages to these so it's clear which one failed without having to look at line numbers